flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriter.java

```diff
@@ -398,6 +407,11 @@ private void registerMetricSync() {
                 });
     }
 
+    private String createProducerClientId(Properties props, int subtaskId) {
+        String prefix = props.getProperty(KafkaSinkOptions.CLIENT_ID_PREFIX.key());
+        return prefix + "-" + subtaskId;
```

Comment:
I think we need to do this for the Commiter as well.

Can we add a string e.g. "writer" and "commiter" to make the client ids distinct?

Reply:
@mas-chen thanks for the suggestion! It doesn't look like it's easy to access a subtask id from the `KafkaCommitter`... Would you recommend using something like a `producerId` or `transactionId` instead? Thanks.

Reply:
It might not be easy to access that since the committer is a global singleton, so you wouldn't run into the need to distinguish between subtask ids

Reply:
Right, updated!

Reply:
Well actually the parallelism of the committer is the same as the writer in Kafka sink instead of 1, so maybe we have to randomize the id here to avoid collision :-(